kiwi/xml_state.py

```diff
@@ -75,6 +75,20 @@ def get_users_sections(self):
             self.xml_data.get_users()
         )
 
+    def get_description_section(self):
+        """
+        The description section of this appliance
+
+        :return: list of <description> section reference(s)
+
+        :rtype: list
+        """
+        # FIXME: why does this break the unit tests?
```

Comment:
@schaefi For some reason the commented out part breaks the unit test I added. Do you have an idea why?

Reply:
yes because the flake8 tests doesn't like the comment type and probably the FIXME :)
You can check yourself with running:

```
make flake
```

which does

```
tox -e check
```

in your tox env

Reply:
flake 8 does not complain, I get the following error when calling `_profiled` on the value that I am currently returning:
```
[gw8] linux -- Python 3.7.2 /home/dan/suse/github.com/SUSE/kiwi/.tox/3.7/bin/python

self = <unit.xml_state_test.TestXMLState object at 0x7ff6ed9a07f0>

    def test_get_description_section(self):
>       assert self.state.get_description_section()\
                         .get_specification()[0]\
                         .strip() == \
            "openSUSE 13.2 JeOS, is a small text based image"

xml_state_test.py:399: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../kiwi/xml_state.py:88: in get_description_section
    self.xml_data.get_description()[0]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <kiwi.xml_state.XMLState object at 0x7ff6ed941da0>
xml_abstract = <kiwi.xml_parse.description object at 0x7ff6ed942240>

    def _profiled(self, xml_abstract):
        """
        return only those sections matching the instance stored
        profile list from the given XML abstract. Sections without
        a profile are wildcard sections and will be used in any
        case
        """
        result = []
>       for section in xml_abstract:
E       TypeError: 'description' object is not iterable

../../kiwi/xml_state.py:1963: TypeError
```

Without the `_profiled` call, everything works fine.

Reply:
yes you are right. This was my fault. I thought the description section also takes a profile attribute but it doesn't. It exists only once and can not be profiled intentionally. For one of the next PR's I also will need access to the description section. Thus I coded my thoughts here: #967